### PR TITLE
SqlBulkTruncation

### PR DESCRIPTION
**Pull request by JRahnama**
This PR adds SqlBulkTruncation Tests.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted.TestFixtures.Setup
+{
+    public class BulkCopyTruncationTables : Table
+    {
+        private const string ColumnEncryptionAlgorithmName = @"AEAD_AES_256_CBC_HMAC_SHA_256";
+        private ColumnEncryptionKey columnEncryptionKey;
+
+
+        public BulkCopyTruncationTables(string tableName, ColumnEncryptionKey columnEncryptionKey1, ColumnEncryptionKey columnEncryptionKey2) : base(tableName)
```
Comment (karinazhou) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs]: The `columnEncryptionKey2` is not used.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs [hunk 2]
```diff
@@ -96,6 +127,110 @@ protected List<Table> CreateTables(IList<ColumnEncryptionKey> columnEncryptionKe
             SqlNullValuesTable = new SqlNullValuesTable(GenerateUniqueName("SqlNullValuesTable"), columnEncryptionKeys[0]);
             tables.Add(SqlNullValuesTable);
 
+            TabNVarCharMaxSource = new BulkCopyTruncationTables(GenerateUniqueName("TabNVarCharMaxSource"), columnEncryptionKeys[0], columnEncryptionKeys[1]);
```
Comment (karinazhou) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs]: We only need 1 columnEncryptionKey.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs [hunk 3]
```diff
@@ -112,8 +247,8 @@ public void Dispose()
                     databaseObjects.ForEach(o => o.Drop(sqlConnection));
                 }
             }
-            
-           
+
```
Comment (karinazhou) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs]: Extra lines.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategyCertStoreProvider.cs [hunk 4]
```diff
@@ -29,6 +29,9 @@ internal override void SetupDatabase()
             databaseObjects.AddRange(columnEncryptionKeys);
 
             List<Table> tables = CreateTables(columnEncryptionKeys);
+
+            Dictionary<string, string> truncationTableNames = sqlBulkTruncationTableNames;
```
Comment (karinazhou) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategyCertStoreProvider.cs]: Where is this `truncationTableNames` used?

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs (hunk 1 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs]: I took them out.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs (hunk 3 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs]: removed.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 5]
```diff
@@ -0,0 +1,522 @@
...
+
+        private readonly Dictionary<string, string> tableNames = new Dictionary<string, string>();
+
+        public SqlBulkCopyTruncation(SQLSetupStrategyCertStoreProvider fixture)
+        {
+            this._fixture = fixture;
+            this.tableNames = fixture.sqlBulkTruncationTableNames;
+        }
+
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.IsEnclaveEnabledAndConnectionStringsProvided))]
```
Comment (karinazhou) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: Why are these tests run for EnclaveEnabled only?

Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: The tables are all encrypted by "RANDOMIZED" encrypotion type. There are tests to check failure  when enclave is enabled on the source and not enalbed on the target.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs [hunk 6]
```diff
@@ -1812,52 +1812,6 @@ public void TestSqlCommandSequentialAccessCodePaths(string connection, CommandBe
 
         }
 
-        [SkipOnTargetFramework(TargetFrameworkMonikers.Netcoreapp)]
-        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringSetupForAE))]
-        [ClassData(typeof(AEConnectionStringProvider))]
-        public void TestExecuteXmlReader(string connection)
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs]: Why are we removing this test?

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 7]
```diff
@@ -0,0 +1,522 @@
+﻿using System;
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: Don't forget the license header.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 8]
```diff
@@ -0,0 +1,522 @@
...
+
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.IsEnclaveEnabledAndConnectionStringsProvided))]
+        [ClassData(typeof(AEConnectionStringProvider))]
+        public void BulkCopyTestsInt(string connectionString)
+        {
+            PopulateTable("TabIntSource", "1, 300", connectionString);
+
+            Assert.Throws<InvalidOperationException>(() => DoBulkCopy(tableNames["TabIntSource"], tableNames["TabTinyIntTarget"], connectionString));
+
+            //Truncate removes the data but not the table.
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: I see this pattern of cleaning up/truncating the tables at the end of the test. But if the test fails, the tables will not be truncated. It looks like you need to put the cleanup in a finally block.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 9]
```diff
@@ -0,0 +1,522 @@
...
+
+            VerifyTablesEqual(tableNames["TabIntSourceDirect"], tableNames["TabIntTargetDirect"], connectionString);
+
+            //Truncate populated tables.
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+                SilentRunCommand($@"TRUNCATE TABLE [{tableNames["TabIntSourceDirect"]}]", connection);
+                SilentRunCommand($@"TRUNCATE TABLE [{tableNames["TabIntTargetDirect"]}]", connection);
+
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: Nit: superfluous extra line.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 10]
```diff
@@ -0,0 +1,522 @@
...
+            Assert.Throws<InvalidOperationException>(() => { DoBulkCopyDirect(tableNames["TabIntSourceDirect"], tableNames["TabIntTargetDirect"], connectionString, false, true); });
+
+            //Truncate populated tables.
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+                SilentRunCommand($@"TRUNCATE TABLE [{tableNames["TabIntSourceDirect"]}]", connection);
+                SilentRunCommand($@"TRUNCATE TABLE [{tableNames["TabIntTargetDirect"]}]", connection);
+
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: Nit: Superfluous extra line.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 11]
```diff
@@ -0,0 +1,522 @@
...
+                    using (SqlDataReader reader = cmd.ExecuteReader())
+                    {
+
+                        while (reader.Read())
+                        {
+                            string columnValue = reader.GetString(0);
+                            Assert.True(columnValue.Equals(new string('a', 8000)), "Unexpected value read");
+                        }
+                    }
+                }
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: It looks like you are missing cleanup/truncate commands here.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 12]
```diff
@@ -0,0 +1,522 @@
...
+        public void Dispose()
+        {
+            foreach (string connection in DataTestUtility.AEConnStringsSetup)
+            {
+                using (SqlConnection sqlConnection = new SqlConnection(connection))
+                {
+                    sqlConnection.Open();
+                    foreach (string val in tableNames.Values)
+                    {
+                        Table.DeleteData(val, sqlConnection);
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: Why do we need to delete the data on Dispose? Doesn't the fixture just drop all these tables when it is disposed?

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs [hunk 13]
```diff
@@ -21,7 +22,37 @@ public class SQLSetupStrategy : IDisposable
...
+        public Table TabSmallBinaryMaxTarget { get; private set; }
+        public Table TabVarBinaryTarget { get; private set; }
+        public Table TabBinaryMaxSource { get; private set; }
+        public Table TabBinaryTarget { get; private set; }
+        public Table TabSmallBinarySource { get; private set; }
+        public Table TabSmallBinaryTarget { get; private set; }
+        public Table TabSmallCharSource { get; private set; }
+        public Table TabSmallCharTarget { get; private set; }
+
+
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs]: Nit: Two superfluous empty lines.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategyCertStoreProvider.cs [hunk 14]
```diff
@@ -11,7 +11,7 @@ public class SQLSetupStrategyCertStoreProvider : SQLSetupStrategy
     {
         public SqlColumnEncryptionCertificateStoreProvider CertStoreProvider;
         public CspColumnMasterKey CspColumnMasterKey;
-
+
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategyCertStoreProvider.cs]: Nit: Trailing spaces (the blank line is okay).

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs [hunk 15]
```diff
@@ -0,0 +1,140 @@
+﻿using System;
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs]: Don't forget the license header.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+using Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted.Setup;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests.AlwaysEncrypted.TestFixtures.Setup
+{
+    public class BulkCopyTruncationTables : Table
+    {
+        private const string ColumnEncryptionAlgorithmName = @"AEAD_AES_256_CBC_HMAC_SHA_256";
+        private ColumnEncryptionKey columnEncryptionKey;
+
+
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs]: Nit: Superfluous empty line.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs [hunk 17]
```diff
@@ -32,7 +32,7 @@ public static class DataTestUtility
         public static readonly string AKVClientSecret = null;
         public static List<string> AEConnStrings = new List<string>();
         public static List<string> AEConnStringsSetup = new List<string>();
-        public static readonly bool EnclaveEnabled = false;
+        public static bool EnclaveEnabled = false;
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs]: Why this change? I don't see other changes where the value is being updated and similar variables right below are still readonly. So wondering...

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs (hunk 6 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs]: Test has been declared as false test. "FOR XML" clause is not supported in AE.

Comment (David-Engel) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs]: It's okay to test for unsupported scenarios. Is the test not passing consistently?

Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs]: Good point David. I can actually modify it to check for the error message that the mentioned clause is not supported on AE.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs (hunk 2 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs]: removed.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategyCertStoreProvider.cs (hunk 4 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategyCertStoreProvider.cs]: It was added by mistake removed.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 5 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: fixed to work on AE

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs (hunk 6 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/ApiShould.cs]: reverted.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 8 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: Add to the applicable parts.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 10 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: removed.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs (hunk 13 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs]: removed.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs (hunk 15 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs]: added.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs (hunk 17 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs]: reverted.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 7 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: added.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategyCertStoreProvider.cs (hunk 14 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategyCertStoreProvider.cs]: removed.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs (hunk 16 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/BulkCopyTruncationTables.cs]: removed.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 9 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: removed.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 18]
```diff
@@ -0,0 +1,697 @@
...
+                            Assert.Equal(10, Convert.ToInt32(datetime.Month));
+                            Assert.Equal(23, Convert.ToInt32(datetime.Day));
+
+                            // To verify the milliseconds, we need to look at the timespan
+                            string millisec = timespan.ToString();
+                            Assert.True(millisec.Equals("12:45:37.1200000"), "Unexpected milliseconds");
+                        }
+                    }
+
+                    SilentRunCommand($@"TRUNCATE TABLE [{tableNames["TabDatetime2Source"]}]", connection);
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: This should be in a finally block.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 19]
```diff
@@ -0,0 +1,697 @@
...
+                connection.Open();
+                using (SqlDataReader reader = cmd.ExecuteReader())
+                {
+                    while (reader.Read())
+                    {
+                        Assert.True(reader.GetString(0).Equals("ab"), "String was not truncated as expected!");
+                    }
+                }
+
+                SilentRunCommand($@"TRUNCATE TABLE [{tableNames["TabVarCharTarget"]}]", connection);
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: This should be in a finally block.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 20]
```diff
@@ -0,0 +1,697 @@
...
+                connection.Open();
+                using (SqlDataReader reader = cmd.ExecuteReader())
+                {
+                    while (reader.Read())
+                    {
+                        Assert.True(reader.GetString(0).Equals(new string('a', 7000)), "String was not truncated as expected!");
+                    }
+                }
+
+                SilentRunCommand($@"TRUNCATE TABLE [{tableNames["TabVarCharMaxSource"]}]", connection);
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: This should be in a finally block.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 21]
```diff
@@ -0,0 +1,697 @@
...
+                            byte[] columnValue = (byte[])reader[0];
+                            Assert.True(3000 == columnValue.Length, "Unexpected array length!");
+
+                            foreach (byte b in columnValue)
+                            {
+                                Assert.Equal(0xee, b);
+                            }
+                        }
+                    }
+                    SilentRunCommand($@"TRUNCATE TABLE [{tableNames["TabBinaryMaxSource"]}]", connection);
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: This should be in a finally block.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs [hunk 22]
```diff
@@ -0,0 +1,697 @@
...
+                {
+                    using (SqlDataReader reader = cmd.ExecuteReader())
+                    {
+                        while (reader.Read())
+                        {
+                            var columnValue = reader.GetString(0);
+                            Assert.True(columnValue.Equals(new string('a', 8000)), "Unexpected value read");
+                        }
+                    }
+                    SilentRunCommand($@"TRUNCATE TABLE [{tableNames["TabBinaryMaxSource"]}]", conn);
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: This should be in a finally block.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 18 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: done.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 20 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: done.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 21 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: done.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 22 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: done.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 19 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: done.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs (hunk 12 above)
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlBulkCopyTruncation.cs]: It was a duplication of IDisposable in couple of AE tests. Removed from test classes. SqlSetupStrategy already implements IDisposable.
